dotnet/Services/CybersourcePaymentService.cs

```diff
@@ -1388,83 +1696,120 @@ public async Task<List<MerchantDefinedInformation>> GetMerchantDefinedInformatio
-                                                            _context.Vtex.Logger.Warn("GetMerchantDefinedInformation", null, $"Invalid operation '{operation}'");
-                                                            break;
+
+                                                                break;
+                                                            case "EQUALS":
+                                                                bool match = string.Equals(propValue, valueSubStrArr[2], StringComparison.OrdinalIgnoreCase);
+                                                                propValue = match.ToString();
+
+                                                                break;
+                                                            default:
+                                                                _context.Vtex.Logger.Warn("GetMerchantDefinedInformation", null, $"Invalid operation '{operation}'");
+                                                                break;
+                                                        }
                                                     }
                                                 }
-                                            }
-                                            else
-                                            {
-                                                var propValueObj = this.GetPropertyValue(requestWrapper, valueSubStr);
-                                                if(propValueObj != null)
+                                                else
                                                 {
-                                                    propValue = propValueObj.ToString();
+                                                    var propValueObj = this.GetPropertyValue(requestWrapper, valueSubStr);
+                                                    if (propValueObj != null)
+                                                    {
+                                                        propValue = propValueObj.ToString();
+                                                    }
                                                 }
                                             }
-                                        }
 
-                                        merchantDefinedValue = merchantDefinedValue.Replace($"{startCharacter}{originalValueSubStr}{endCharacter}", propValue);
-                                        sanityCheck = sanityCheck + 1;
+                                            merchantDefinedValue = merchantDefinedValue.Replace($"{startCharacter}{originalValueSubStr}{endCharacter}", propValue);
+                                            sanityCheck = sanityCheck + 1;
+                                        }
+                                        while (merchantDefinedValue.Contains(startCharacter) && merchantDefinedValue.Contains(endCharacter) && sanityCheck < 100);
```

Comment:
I don't think this while loop is doing anything?

Reply:
If there is more than one replacement on a line, it loops until there are no more to be replaced.